src/adams_methods.jl

```diff
@@ -0,0 +1,636 @@
+            # the last 'order' number of points
+            steporder = n==1 ? 1 : n-1
+
+        elseif adaptive_order && steporder>1
+            # First, compute truncation errors for implicit orders: k-1,k
+            # against current k+1
+            for i = 1:dof
+                yerr_km1[i] = dt*(g[steporder] - g[steporder-1])*ϕ[3,steporder][i]
+            end
+            err_km1 = stepsize_hw92!(dt, tdir, y[n], y_np1, yerr_km1, steporder,
+                            timeout_VS,dof, abstol, reltol, maxstep, norm)[1]
+
+            for i = 1:dof
+                yerr_k[i] = dt*(g[steporder+1] - g[steporder])*ϕ[3,steporder+1][i]
+            end
+            err_k = stepsize_hw92!(dt, tdir, y[n], y_np1, yerr_k, steporder,
+                            timeout_VS, dof, abstol, reltol, maxstep, norm)[1]
+
+            # Decrement steporder if LE_{k} and LE_{k-1} < LE_{k+1}
+            if max(err_km1,err_k) <= err_kp1
+                steporder = max(order,steporder - 1)
+
+            elseif consec_const_steps >= steporder && timeout_VO==0 && n>steporder + 3
+                # Given enough consecutive steps taken at a constant step size
+                # we compute truncation error for order implicit k+2
+                ϕ_and_ϕstar_coefs!(b, ϕ, ϕstar,dy[n],t,t_np1,n,steporder+2)
+                ϕ_coefs!(ϕ, ϕstar,dy_np1,steporder+3)
+
+                for i = 1:dof
+                    yerr_kp2[i] = dt*γ_star(steporder + 3)*ϕ[3,steporder+3][i]
+                end
+                err_kp2 = stepsize_hw92!(dt,tdir,y[n],y_np1,yerr_kp2,steporder,
+                            timeout_VS, dof, abstol, reltol, maxstep, norm)[1]
+
+                # Increment steporder if LE_{k+2} < LE_{k+1} and max(LE_{k},
+                # LE_{k-1}) > LE_{k+1}
+                if err_kp2 < err_kp1
+                    steporder = min(max_order,steporder + 1)
+                    timeout_VO = timeout_VO_const # reset timout_VO
+                end
+            end
+            # Keep timeout_VO 0, or decrement
+            timeout_VO= timeout_VO==0? 0 : timeout_VO-1
+        end # end of adaptive order selection
+
+        #--#####################################################################
+        # Prepare for next step
+        #--#####################################################################
+
+        # Shift data from this step, from the "current step" position to the
+        # "previous step" position for ϕstar and ϕ
+        # Note: ϕ[2,:] and ϕ_star[2,:] corresponds to ϕ_n and ϕ*_n, and
+        # ϕ_star[1,:] to ϕ*_{n-1}, etc
+        # TODO: implement with DataStructures.CircularBuffer
+        ϕstar[1,1:min(max(n-1,1),current_steporder+1)] =
+                                ϕstar[2,1:min(max(n-1,1),current_steporder+1)]
+
+        # Update stepsize variables
+        prev_dt = dt
+        dt = next_dt
+        # Increment our step
+        n= n+1
+    end #while loop over steps
+
+    if points == :steps
```

Comment:
Yeah, I wanted to ask you about this. So, currently the options for `points` are 

- `specified`
- `all`, which is the union of points `specified` and points which were used to step through the time range

I thought, it might be worthwhile to also have a `steps` options which is `all` - `specified`; namely, the points actually used to calculate the solution. Though, it is mostly  because it is very easy to add this extra option.